units/en/unit3/hands-on.mdx

```diff
@@ -0,0 +1,320 @@
+
+## Set the GPU 💪
+- To **accelerate the agent's training, we'll use a GPU**. To do that, go to `Runtime > Change Runtime type`
+
+<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/notebooks/gpu-step1.jpg" alt="GPU Step 1">
+
+- `Hardware Accelerator > GPU`
+
+<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/notebooks/gpu-step2.jpg" alt="GPU Step 2">
+
+## Create a virtual display 🔽
+
+During the notebook, we'll need to generate a replay video. To do so, with colab, **we need to have a virtual screen to be able to render the environment** (and thus record the frames).
+
+Hence the following cell will install the librairies and create and run a virtual screen 🖥
+
+```bash
+apt install python-opengl
+apt install ffmpeg
+apt install xvfb
+pip3 install pyvirtualdisplay
+```
+
+```bash
+apt-get install swig cmake freeglut3-dev
+```
+
+```bash
+pip install pyglet==1.5.1
+```
+
+```python
+# Virtual display
+from pyvirtualdisplay import Display
+
+virtual_display = Display(visible=0, size=(1400, 900))
+virtual_display.start()
+```
+
+## Clone RL-Baselines3 Zoo Repo 📚
+You can now directly install from python package `pip install rl_zoo3` but since we want **the full installation with extra environments and dependencies** we're going to clone `RL-Baselines3-Zoo` repository and install from source.
+
+```bash
+git clone https://github.com/DLR-RM/rl-baselines3-zoo
+```
+
+## Install dependencies 🔽
+We can now install the dependencies RL-Baselines3 Zoo needs (this can take 5min ⏲)
+
+But we'll also install:
+- `huggingface_sb3`: Additional code for Stable-baselines3 to load and upload models from the Hugging Face 🤗 Hub.
+
+```bash
+cd /content/rl-baselines3-zoo/
+```
+
+```bash
+pip install -r requirements.txt
+```
+
+```bash
+pip install huggingface_sb3
+```
+
+## Train our Deep Q-Learning Agent to Play Space Invaders 👾
+
+To train an agent with RL-Baselines3-Zoo, we just need to do two things:
+1. We define the hyperparameters in `rl-baselines3-zoo/hyperparams/dqn.yml`
+
+<img src="https://huggingface.co/datasets/huggingface-deep-rl-course/course-images/resolve/main/en/notebooks/unit3/hyperparameters.png" alt="DQN Hyperparameters">
+
+
+Here we see that:
+- We use the `Atari Wrapper` that preprocess the input (Frame reduction ,grayscale, stack 4 frames)
+- We use `CnnPolicy`, since we use Convolutional layers to process the frames
+- We train it for 10 million `n_timesteps`
+- Memory (Experience Replay) size is 100000, aka the amount of experience steps you saved to train again your agent with.
+
+💡 My advice is to **reduce the training timesteps to 1M,** which will take about 90 minutes on a P100. `!nvidia-smi` will tell you what GPU you're using. At 10 million steps, this will take about 9 hours, which could likely result in Colab timing out. I recommend running this on your local computer (or somewhere else). Just click on: `File>Download`.
```

Comment:
Modifying this file tells me i cannot save a copy in drive and force me to create a copy of the unit 3 colab, which creates lots of friction. I would suggest since the beginning to create a copy